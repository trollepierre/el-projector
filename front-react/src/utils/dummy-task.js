const dummyTask = (task) => ({
  id: 1,
  name: 'Build el-projector',
  description: 'more data about how to do it',
  points: 999, // given by server
  createdDate: '19/05/2019',
  endDate: undefined, // must be done before this date
  optimalDate: undefined, // user would do it at this date
  isDone: false,
  doneDate: undefined, // for log record
  priority: 'HIGHLY_IMPORTANT', // Enum with 5 levels (inherited by default, but can be updated)
  benefit: 'ORGASM_HAPPINESS', // Enum with 5 levels
  neededTime: 'ONE_MONTH', // Enum
  epicId: 'A1', // Yearly happiness epic
  neededMaterial: ['MacTo', 'iMoi', 'MacNo', 'home'],
  subTaskIds: ['TASK10', 'TASK11', 'TASK12'], // checklist of subtasks that depends on it
  masterTaskId: 'TASK0', // this task depends on the master one
  ...task,
})

export { dummyTask }
