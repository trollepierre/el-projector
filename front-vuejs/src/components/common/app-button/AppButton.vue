<template>
  <button
    :style="{backgroundColor: backgroundColor, color: color}"
    class="app-button"
    @click="onClick"
  >
    {{ text }}
  </button>
</template>
<script>
  export default {
    name: 'AppButton',
    props: {
      isDisabled: { type: Boolean, default: () => false },
      color: { type: String, default: () => '' },
      backgroundColor: { type: String, default: () => '' },
      text: { type: String, default: () => '' },
    },
    methods: {
      onClick() {
        !this.isDisabled ? this.$emit('click') : null
      },
    },
  }
</script>
<style scoped>
  .app-button {
    width: 85%;
    padding: 8px;
    margin: 3px;
    font-size: 14px;
    font-weight: bold;
    cursor: pointer;
    border: 1px solid black;
    border-radius: 10px;
    box-shadow: 2px 2px 20px 0 rgba(0, 0, 0, 0.1);
    text-transform: uppercase;
  }
</style>
