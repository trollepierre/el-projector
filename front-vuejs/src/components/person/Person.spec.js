import Task from './Task.vue'
import AppButton from '../common/app-button/AppButton.vue'
import { dummyTask } from '../../utils/test/dummy-task'

describe('Task.vue', () => {
  const task = dummyTask()

  it('should match snapshot', () => {
    // When
    const wrapper = shallowMount(Task, { propsData: { task } })

    // Then
    expect(wrapper.element).toMatchSnapshot()
  })

  describe('events', () => {
    it('should meet on click on seen button', () => {
      // Given
      const wrapper = shallowMount(Task, { propsData: { task } })

      // When
      wrapper.findAll(AppButton).at(0).vm.$emit('click')

      // Then
      expect(wrapper.emitted('meet')[0][0]).toEqual(task)
    })

    it('should differ meeting on click on later button', () => {
      // Given
      const wrapper = shallowMount(Task, { propsData: { task } })

      // When
      wrapper.findAll(AppButton).at(1).vm.$emit('click')

      // Then
      expect(wrapper.emitted('differ')[0][0]).toEqual(task)
    })

    it('should reject meeting on click on nope button', () => {
      // Given
      const wrapper = shallowMount(Task, { propsData: { task } })

      // When
      wrapper.findAll(AppButton).at(2).vm.$emit('click')

      // Then
      expect(wrapper.emitted('reject')[0][0]).toEqual(task)
    })
  })
})
