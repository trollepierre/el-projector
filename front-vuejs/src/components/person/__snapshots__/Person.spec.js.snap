// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task.vue should match snapshot 1`] = `
<tr>
  <td
    class="next"
  >

    05/11/1996

  </td>

  <td
    class="first-name"
  >

    Arthur

  </td>

  <td
    class="last-name"
  >

    King

  </td>

  <td
    class="points"
  >

    999

  </td>

  <td
    class="adjusted-points"
  >

    800

  </td>

  <td
    class="last-date"
  >

    04/11/1996

  </td>

  <td
    class="impact"
  >
    <appbutton-stub
      backgroundcolor="green"
      color="white"
      text="seen"
    />

    <appbutton-stub
      backgroundcolor="orange"
      color="white"
      text="later"
    />

    <appbutton-stub
      backgroundcolor="red"
      color="white"
      text="nope"
    />
  </td>

  <td
    class="birth-date"
  >



  </td>

  <td
    class="origin"
  >



  </td>

  <td>
    <appbutton-stub
      backgroundcolor=""
      color=""
      text="edit"
    />

    <appbutton-stub
      backgroundcolor=""
      color=""
      text="delete"
    />
  </td>
</tr>
`;
